source/common/http/http1/conn_pool.cc

```diff
@@ -147,7 +151,12 @@ void ConnPoolImpl::onConnectionEvent(ActiveClient& client, Network::ConnectionEv
     } else if (!client.connect_timer_) {
       // The connect timer is destroyed on connect. The lack of a connect timer means that this
       // client is idle and in the ready pool.
-      removed = client.removeFromList(ready_clients_);
+      if (client.delayed_) {
+        client.delayed_ = 0;
```

Comment:
nit: it's not required

Reply:
I would prefer to maintain the invariant that client.delayed_ is true IFF it is in the delayed_clients_ list rather than depending on removed being deleted, e.g. a better fix for our problem would be to handle half-close correctly which would mean that 'removed' might not be deleted immediately.